### DISCUSSION TITLE
Extend function subset and the expression parser with support for Map

### DISCUSSION THREAD

**Discussion by josdejong** [josdejong/mathjs#3046] [Ideas]
Mathjs now supports using a `Map` as `scope`. It would be nice if it is possible to use `Map` in variables too.

```js
const a = new Map([
  ['b', 42]
])

math.subset(a, ['b'])     
// TypeError: Unexpected type of argument in function subset 
//   (expected: string or Array or Matrix or Object or number or boolean, actual: Map, index: 0)

const scope = new Map([
  ['a', a]
])
math.evaluate('a.b', scope) 
// Error: No access to property "b" 
```

A first step would be to be able to use externally created Maps in the function `subset` and in the expression parser. A second step would be to be able to configure mathjs to either create a plain Object or a Map in the expression parser when a user evaluates something like `math.evaluate('{ b : 42 }')`.

Help would be welcome! @jhugman would this be something you would like to implement? (feel free to say no)

**Comment by jhugman** [comment]
I haven't got an awful lot of bandwidth now for a little while, but if this isn't high priority, I can chip away at it.

**Comment by josdejong** [comment]
👍 no worries, please only pick it up if you like to. Let's see who picks it up first. Just drop a message here if you're in the mood and _do_ have time for it so we prevent multiple people accidentally implement the same thing at the same time.

**Comment by gwhitney** [comment]
As this seems to be an open-ended proposal to add Map as a type that mathjs can natively handle, rather than a specific actionable issue (and because if I am interpreting the proposal correctly, it is probably lower priority than adding bigint support), moving to a Discussion.
